pulumi/README.md

```diff
@@ -0,0 +1,96 @@
+# @operator-error/pulumi-lambda-cert
+
+_This is a package containing components for use with [Pulumi][pulumi], a tool for provisioning cloud infrastructure 
+based on a description written with general purpose programming languages._
+
+This package provides a component named `LambdaCert`, which can be used to create the resources
+needed to run [Lambda Cert][lambdacert], a function for maintaining Let's Encrypt certificates using
+AWS Lambda.
+
+## Usage
+
+```typescript
+import { LambdaCert, LambdaCertOutputs, letsEncryptV2ProductionUrl } from "@operator-error/pulumi-lambda-cert"
+
+async function main(): Promise<LambdaCertOutputs> {
+	return await LambdaCert.create("vault", {
+		// A Description of the purpose of this configuration (e.g. "Vault", "HAProxy"). Used in Tags.
+		description: "Vault",
+
+		// These tags will be applied to every object which supports tagging.
+		// Avoid using `Name`, since this will be overridden by many components.
+		baseTags: {
+			Project: "Staging Vault",
+		},
+
+		// The part of the domain name for which a certificate is required which does _not_
+		// form part of the hosted zone name. For example, if requesting a certificate for
+		// `*.vault.operator-error.com`, and the public hosted zone is for `operator-error.com`,
+		// `domainNamePrefix` should be set to `*.vault` (without a trailing `.`).
+		domainNamePrefix: "vault-staging",
+		// The part of the domain name for which a certificate is required which is included
+		// in the hosted zone name.
+		route53DomainName: "operator-error.com",
+
+		// The admin email address to provide to the ACME service when creating an account.
+		adminEmail: "webmaster@operator-error.com",
+
+		// The URL of the ACME server. Defaults to Let's Encrypt ACME V2 Staging. Constants
+		// are exported from the package for both Let's Encrypt staging and production.
+		acmeUrl: letsEncryptV2ProductionUrl,
+
+		// If set to true, add the certificate to an Java KeyStore encrypted with the same
+		// KMS key as the private key, and uploaded to the bucket along with the certificate
+		// and encrypted private key.
+		generateJavaKeyStore: true,
+
+		// If set to true, force delete the entire contents of the bucket when destroying
+		// the component using Pulumi. This can be useful for quick iteration in development
+		// when using a staging endpoint, though be wary of rate limits when using a real
+		// ACME service.
+		emptyBucketOnDestroy: true,
+	});
+}
+
+const outputs = main();
+export const keyBucketArn = outputs.then(o => o.keyBucketArn);
+export const keyBucketName = outputs.then(o => o.keyBucketName);
+export const kmsKeyArn = outputs.then(o => o.kmsKeyArn);
+export const kmsKeyId = outputs.then(o => o.kmsKeyId);
```

Comment:
I wish we had a better way of doing this sort of thing with less boilerplate 🙁

Reply:
I opened an issue with a suggestion which _may_ work [here](https://github.com/pulumi/pulumi/issues/1611) - I haven't looked at implementing it yet though. It would definitely be nice to eliminate the boilerplate here.